src/main/app/forms/validation/validators/isLessThanOrEqualToSumOf.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface
+} from 'class-validator'
+
+@ValidatorConstraint()
+export class IsLessThanOrEqualToSumOfConstraint implements ValidatorConstraintInterface {
+  validate (value: any, args?: ValidationArguments) {
+    if (value === undefined) {
+      return true
+    }
+    const argsLength = args.constraints.length
+    const expectedNumberOfArgs = 2
+    if (argsLength !== expectedNumberOfArgs) {
+      throw new Error(`Invalid number of arguments, got ${argsLength}, expected ${expectedNumberOfArgs}`)
+    }
+
+    const amountToSumProperty = args.constraints[0]
+    const lessThanAmountPropertyName = args.constraints[1]
+
+    const lessThanAmount = (args.object as any)[lessThanAmountPropertyName]
+    const amountToSum = (args.object as any)[amountToSumProperty]
+
+    // Not this validator's concern if the amount hasn't been entered
+    if (!amountToSum) {
```

Comment:
Is returning valid in case value exceeds `lessThanAmount` but `amountToSum ` has not been specified correct. I would more expect `amountToSum` to fallback to `0` in such scenario which would simply ignore it but validator would still asses that value is less then max limit.

Reply:
Nope, the fields also have a:
```
@IsPositive({ message: ValidationErrors.INSTALMENTS_AMOUNT_INVALID })
```
Which marks the fields as required basically.
If the above check isn't there then validation gets failed on both fields when it should only fail on the missing field